float64 velL
float64 velR